sqlalchemy-stubs/engine/interfaces.pyi

```diff
@@ -19,17 +20,20 @@ from ..exc import StatementError
 from ..sql.compiler import Compiled as Compiled
 from ..sql.compiler import IdentifierPreparer
 from ..sql.compiler import TypeCompiler as TypeCompiler
+from ..sql.schema import Column
 from ..sql.type_api import TypeEngine
 
-class _OnConnect(Protocol):
-    def __call__(self, connection: Connection) -> None: ...
-
-class _DBAPIConnection(Protocol):
+class _DBAPIConnectionBase(Protocol):
     def close(self) -> None: ...
     def commit(self) -> None: ...
     def cursor(self) -> _DBAPICursor: ...
 
-class _DBAPICursor(Protocol):
+class _DBAPIConnectionTransactions(_DBAPIConnectionBase, Protocol):
+    def rollback(self) -> None: ...
+
+_DBAPIConnection = Union[_DBAPIConnectionBase, _DBAPIConnectionTransactions]
```

Comment:
this is correct, but I'm not sure if it would just annoy people that have to use the `!` ts equivalent in python to ignore the possibly undefined access.

Bug we can tweak it later if we need to